File: scripts/expand_competition_exclusions.R
#!/usr/bin/env Rscript
# Write config/excluded_columns_expanded.txt (one name per line).

cmd <- commandArgs(trailingOnly = FALSE)
fn <- sub("^--file=", "", cmd[grep("^--file=", cmd)])
root <- normalizePath(file.path(dirname(fn), ".."), winslash = "/", mustWork = TRUE)
setwd(root)
source(file.path(root, "R", "meps_competition_exclusions.R"))

out_path <- file.path(root, "config", "excluded_columns_expanded.txt")
lines <- sort(meps_expanded_exclusion_names())
dir.create(dirname(out_path), showWarnings = FALSE, recursive = TRUE)
con <- file(out_path, open = "wt", encoding = "UTF-8")
writeLines(
  c(
    "# Expanded competition-excluded MEPS-style names (physical PUF columns).",
    "# Includes PERWTyyF, VARSTR, VARPSU, BRR1..BRR128, Section 2.5.11 stems + yy.",
    ""
  ),
  con
)
writeLines(lines, con)
close(con)
message("Wrote ", out_path, " (", length(lines), " names)")

File: legacy/python/expand_competition_exclusions.py
#!/usr/bin/env python3
"""Write config/excluded_columns_expanded.txt from src/meps_competition_exclusions.py."""

import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(ROOT))

from src.meps_competition_exclusions import write_expanded_list  # noqa: E402


def main() -> None:
    out = ROOT / "config" / "excluded_columns_expanded.txt"
    out.parent.mkdir(parents=True, exist_ok=True)
    write_expanded_list(str(out))
    print(f"Wrote {out}")


if __name__ == "__main__":
    main()
